backend/remote-state/azure/client.go

```diff
@@ -67,6 +70,17 @@ func (c *RemoteClient) Put(data []byte) error {
 	}
 
 	ctx := context.TODO()
+
+	if c.versioning {
+		snapshotInput := blobs.SnapshotInput{LeaseID: options.LeaseID}
+
+		if _, err := c.giovanniBlobClient.Snapshot(ctx, c.accountName, c.containerName, c.keyName, snapshotInput); err != nil {
+			return err
```

Comment:
could we wrap this error to be a little clearer here:

```suggestion
			return fmt.Errorf("Error snapshotting Blob %q (Container %q / Account %q): %+v", c.keyName, c.containerName, c.accountName, err)
```